VideoLocker/src/main/java/org/edx/mobile/base/BaseSingleFragmentActivity.java

```diff
@@ -171,4 +166,10 @@ protected void hideMessageInSitu(){
         }
     }
 
+    private void showErrorDialog(String message) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage(message)
+                .setPositiveButton(R.string.label_ok, null)
+                .create().show();
```

Comment:
Dialogs need to have their lifecycle managed for proper binding with the Activity, to ensure their retention and avoid leaking as @bguertin pointed out. Originally the `Activity` implementation defined callback methods for managing them (i.e. `onCreateDialog()`, `onPrepareDialog()`, `showDialog()`, `dismissDialog()`). These methods were deprecated in Honeycomb with the introduction of `DialogFragment`, which became the recommended mechanism for managing the lifecycle of dialogs. So basically, there's nothing wrong with using `AlertDialog`; but to have it managed properly, it should be wrapped in a `DialogFragment`, which can be written simply to just return the `AlertDialog` in the `onCreateDialog()` callback method.